### ISSUE TITLE
estimation of the time it takes

### ISSUE THREAD

**Issue by sariagoudarzvand** [DARIAH-DE/Topics#66]
hello,

Thanks for replying :)
I have run the code and it still running, I was wondering whats your estimation of the time it takes for almost 200mg documents which contain 500K documents?
I thought it should not take time more than 2 days, but now its 3days and still running.
Should mention that its 500g ram actually running on a powerful server.

Do you have any idea?

**Comment by severinsimmler** [comment]
Hi Saria,

topic modeling can take quite a while. This depends, however, on the one hand on the size of the text corpus (as you said, your corpus _is_ very extensive, but definitely not _too_ extensive – in any case a reasonable size for topic modeling), and on the other hand on the **number of iterations**. How many iterations have you set? Actually informative logging should be printed to the CLI (or `nohup.out` file, or whatever), which informs you about the current iteration?

Besides, you should know, that this library supports three different LDA implementations:
* [lda](http://pythonhosted.org/lda/index.html), which is lightweight and provides basic LDA. Compared to the others, **this one is quite slow**.
* [MALLET](http://mallet.cs.umass.edu/), which is known to be very robust, **and pretty fast**.
* [Gensim](https://radimrehurek.com/gensim/), which is attractive because of its multi-core support. This one **should be the fastest**, but we haven't done any tests yet to make a reliable comparison.

I would **recommend MALLET in spite of everything**, (my personal) experience has shown that it produces the best models at an acceptable time. Have a look at [this notebook](https://github.com/DARIAH-DE/Topics/blob/master/notebooks/IntroducingMallet.ipynb).

Good luck!

PS: We would be very happy if you share your experiences with the general usability of this library. We are happy to incorporate feedback into further development :slightly_smiling_face:

**Comment by saria85** [comment]
Hi and thanks for supporting:)
most probably I run mallet as I already knew the lda version takes time, 
I set iteration at 1000, but its good to know that there is logging information so I can track it :)
I m not at office right know once checking tomorrow I will share my experience here. (I hope its finished by tomorrow :|)

thanks again :)

**Comment by saria85** [comment]
sadly I have forgotten to pass implementations=mallet to the method. LDA was running for 4 days and no output so I just stopped it. then I run with mallet it was also running for more than 3 days and still no output. I stopped it. and did some preprocessing and make my data 150K. now Im going to run again.
my first impression is that it is very slow. as an example I gave only 1 iteration with 20 topic on mallet on the 150K data after 30 min its still running! im saying only 1 iteration!.
I will update here.

Thanks. :)
